src/main/kotlin/com/emberjs/hbs/linter/ember-template-lint/TemplateLintExternalRunner.kt

```diff
@@ -0,0 +1,261 @@
+        }
+    }
+
+    fun execute(input: JSLinterInput<TemplateLintState>): JSLinterAnnotationResult? {
+        val fileToLint = input.virtualFile
+        return if (fileToLint.isValid && fileToLint.parent != null) {
+            try {
+                val state = input.state
+                val project = input.project
+                if (project.isDisposed) {
+                    null
+                } else {
+                    val workingDirectory = ReadAction.compute<VirtualFile?, RuntimeException> { TemplateLintUtil.findWorkingDirectory(fileToLint) }
+                    if (workingDirectory == null) {
+                        null
+                    } else {
+                        if (myIsOnTheFly) {
+                            (ServiceManager.getService(project, TemplateLintConfigFileChangeTracker::class.java) as TemplateLintConfigFileChangeTracker).startIfNeeded()
+                        }
+                        val canRun = !myIsOnTheFly || ReadAction.compute<Boolean, RuntimeException> {
+                            if (project.isDisposed) {
+                                return@compute false
+                            } else {
+                                val configs = TemplateLintUtil.getPossibleConfigs(workingDirectory)
+                                return@compute (ServiceManager.getService(project, TemplateLintUnsavedConfigFileManager::class.java) as TemplateLintUnsavedConfigFileManager).requestSaveIfNeeded(configs)
+                            }
+                        } as Boolean
+                        if (!canRun) {
+                            LOG.debug("TemplateLint postponed because of unsaved configs")
+                            null
+                        } else {
+                            val interpreter = state.interpreterRef.resolveNotNull(input.project)
+                            val error = JSLinterUtil.validateInterpreterAndPackage(input.project, interpreter, state.templateLintPackage, TemplateLintUtil.PACKAGE_NAME, fileToLint)
+                            if (error != null) {
+                                JSLinterAnnotationResult.create(input, error, null as VirtualFile?)
+                            } else {
+                                val templateLintPackage = TemplateLintPackage.fromNodePackage(input.project, state.templateLintPackage)
+                                val sessionData = TemplateLintSessionData(interpreter, templateLintPackage, workingDirectory, fileToLint, input.fileContent)
+                                templateLint(input, sessionData)
+                            }
+                        }
+                    }
+                }
+            } catch (var11: ExecutionException) {
+                createFileLevelWarning(var11.message!!, input)
+            }
+        } else {
+            null
+        }
+    }
+
+//    fun execute(input: JSLinterInput<TemplateLintState>): JSLinterAnnotationResult? {
```

Comment:
can we remove the comment?

Reply:
whoops yes sorry. Forgot to remove it after migrating away from the outdated usage.